File: problems/individual_problems/roman_numeral_conversion.py
def romanToInt(s):
        """
        :type s: str
        :rtype: int
        """
        key = {'I': 1, 'V': 5, 'X': 10, 'L': 50, 'C': 100, 'D': 500, 'M': 1000}
        result = 0
        reverstr = s[::-1]
        state = "add"
        for i, ch in enumerate(reverstr):
            value = key[ch]
            if state == "add": result += value 
            if state == "subtract": result -= value 
            
            if i < len(s) - 1:
                state = "add"
                compare = s[i + 1]
                if (compare == "V" or compare == "X") and ch == "I": state = "subtract" 
                if (compare == "L" or compare == "C") and ch == "X": state = "subtract" 
                if (compare == "D" or compare == "M") and ch == "C": state = "subtract" 
        return result

# print(romanToInt("III"))

# num = 700
# base = len(`num`) 
# print(`num`[0])
# print(base)

File: problems/individual_problems/roman_numeral_conversion.js
// roman numerals exercise
var romanToInt = function (s) {
    // store conversion key (e.g., {"I": 1, "V": 5 ...})
    // store a state that is either "add" or "subtract"
    // read from right to left
    // add 1, 5, 10, 50, 100, 500, 1000 depending on key if state is "add"
    // subtract if state = "subtract"
    // change state to "add"
    // set conditionals for changing state to subtract (if current === "X" and next === "I", change state to "subtract")

    let result = 0;
    let key = { I: 1, V: 5, X: 10, L: 50, C: 100, D: 500, M: 1000 };
    let marker = "add"
    for (let i = s.length - 1; i >= 0; i--) {
        let char = s[i];
        let value = key[char];
        if (marker === "add") result += value;
        if (marker === "subtract") result -= value;

        if (i > 0) {
            marker = "add";
            let compareChar = s[i - 1];
            if ((char === "V" || char === "X") && compareChar === "I") marker = "subtract";
            if ((char === "L" || char === "C") && compareChar === "X") marker = "subtract";
            if ((char === "D" || char === "M") && compareChar === "C") marker = "subtract";
        }
    }
    return result;
};

console.log(romanToInt("III"))